### PR TITLE
feat: add new metrics required by release dashboard and SLO

### PR DESCRIPTION
**Pull request by theflockers**
this commit adds new metrics to enable a new SLO and Dashboard for the Release Service

Signed-off-by: Leandro Mendes <lmendes@redhat.com>

### CODE REVIEW

**Comment by theflockers** [comment]
@amisstea following up our conversation on Slack a bit more than a week ago, I am adding some new metrics I need in my dashboard/ SLO alerts. Could you please take a look? Thanks.

**Review by yftacherzog — APPROVED** [review]
/lgtm
/approve

File: components/monitoring/prometheus/base/monitoringstack/monitoringstack.yaml [hunk 1]
```diff
@@ -114,9 +115,12 @@ spec:
         - '{__name__="controller_runtime_reconcile_errors_total", namespace!~".*-tenant|openshift-.*|kube-.*"}'
         - '{__name__="controller_runtime_reconcile_total", namespace!~".*-tenant|openshift-.*|kube-.*"}'
         - '{__name__="kube_pod_status_unschedulable", namespace!~".*-tenant|openshift-.*|kube-.*"}'
-        - '{__name__="kube_pod_container_status_restarts_total", namespace="openshift-pipelines"}'
-        - '{__name__="kube_pod_container_status_waiting_reason", namespace!~".*-tenant|openshift-.*|kube-.*"}'
-        - '{__name__="kube_pod_status_phase", namespace!~".*-tenant|openshift-.*|kube-.*"}'
+        - '{__name__="kube_pod_container_status_restarts_total", namespace="openshift-pipelines|release-service"}'
```
Comment (yftacherzog) [inline components/monitoring/prometheus/base/monitoringstack/monitoringstack.yaml:118]: I think this should have been:
```
        - '{__name__="kube_pod_container_status_restarts_total", namespace=~"openshift-pipelines|release-service"}'
```
or:
```
        - '{__name__="kube_pod_container_status_restarts_total", namespace!~"openshift-pipelines|release-service"}'
```

File: components/monitoring/prometheus/base/monitoringstack/monitoringstack.yaml [hunk 2]
```diff
@@ -114,9 +115,12 @@ spec:
...
+        - '{__name__="kube_pod_container_status_waiting_reason", namespace!~".*-tenant|openshift-.*|kube-.*|release-service"}'
```
Comment (yftacherzog) [inline components/monitoring/prometheus/base/monitoringstack/monitoringstack.yaml:119]: This doesn't return anything in `prd-rh01`. Can you confirm that this is the query you needed?

File: components/monitoring/prometheus/base/monitoringstack/monitoringstack.yaml [hunk 3]
```diff
@@ -114,9 +115,12 @@ spec:
         - '{__name__="controller_runtime_reconcile_errors_total", namespace!~".*-tenant|openshift-.*|kube-.*"}'
         - '{__name__="controller_runtime_reconcile_total", namespace!~".*-tenant|openshift-.*|kube-.*"}'
         - '{__name__="kube_pod_status_unschedulable", namespace!~".*-tenant|openshift-.*|kube-.*"}'
-        - '{__name__="kube_pod_container_status_restarts_total", namespace="openshift-pipelines"}'
-        - '{__name__="kube_pod_container_status_waiting_reason", namespace!~".*-tenant|openshift-.*|kube-.*"}'
-        - '{__name__="kube_pod_status_phase", namespace!~".*-tenant|openshift-.*|kube-.*"}'
```
Comment (yftacherzog) [inline components/monitoring/prometheus/base/monitoringstack/monitoringstack.yaml:119]: @theflockers  The changes introduced to those lines, exclude namespace `release-service` from the result that being forwarded for those metrics. Is that the intention?

**Comment by theflockers** [comment]
@yftacherzog I will answer all at once here. I see clearly there was some misreading of the queries from my part. I will add another PR correcting some of these values, so we can discuss there.

**Comment by yftacherzog** [comment]
> @yftacherzog I will answer all at once here. I see clearly there was some misreading of the queries from my part. I will add another PR correcting some of these values, so we can discuss there.

cool

**Pull request merged**